src/js/structs/PodContainer.js

```diff
@@ -16,21 +17,27 @@ module.exports = class PodContainer extends Item {
           return PodContainerStatus.RUNNING;
         }
 
-      case 'ERROR':
+      case PodContainerState.ERROR:
         return PodContainerStatus.ERROR;
 
-      case 'FAILED':
+      case PodContainerState.FAILED:
         return PodContainerStatus.FAILED;
 
-      case 'FINISHED':
+      case PodContainerState.FINISHED:
         return PodContainerStatus.FINISHED;
 
-      case 'KILLED':
+      case PodContainerState.KILLED:
         return PodContainerStatus.KILLED;
 
       default:
+        let statusString = this.get('status') || '';
+        if (statusString.startsWith('TASK_')) {
+          statusString = statusString.substr(5);
+        }
+        statusString = statusString.replace(/_/g, ' ');
```

Comment:
There are other statuses currently not handled (check this: https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto#L1497 ) - I don't know which one of them can appear on the pods context.